diff --git a/steamdeps b/steamdeps
index cff27d9..317c6da 100755
--- a/steamdeps
+++ b/steamdeps
@@ -34,7 +34,7 @@ def getDistro():
 	Try to identify distribution of Linux
 	"""
 	global _distro
-	if dist()[0].lower() == 'fedora':
+	if dist()[0].lower() == 'fedora' or dist()[0].lower() == 'korora':
 		_distro = 'fedora'
 		_distro_version = dist()[1]
 	elif dist()[0].lower() == 'redhat':
@@ -152,8 +152,15 @@ class Package:
 
 ###
 def hasPackage( package ):
-	process = subprocess.Popen( ['dpkg', '-l', package], stdout=subprocess.PIPE, stderr=subprocess.PIPE )
-	installed_pattern = re.compile( r"^\Si\s+([^\s]+)\s+([^\s]+)" )
+	distro = getDistro()
+
+	if distro == 'fedora' or distro == 'rhel' or distro == 'suse':
+		process = subprocess.Popen( ['rpm', '-q', package], stdout=subprocess.PIPE, stderr=subprocess.PIPE )
+		installed_pattern = re.compile( r"^((?!package).*)$" )
+	elif distro == 'ubuntu' or distro == 'debian' or distro == 'unknown':
+		process = subprocess.Popen( ['dpkg', '-l', package], stdout=subprocess.PIPE, stderr=subprocess.PIPE )
+		installed_pattern = re.compile( r"^\Si\s+([^\s]+)\s+([^\s]+)" )
+
 	for line in process.stdout:
 		line = line.decode( "utf-8" ).strip()
 		match = re.match( installed_pattern, line )
@@ -357,11 +364,18 @@ def checkConfig( config ):
 		sys.stderr.write( "Unsupported dependency version: %s\n" % config["STEAM_DEPENDENCY_VERSION"] )
 		return False
 
+	distro = getDistro()
+
+        package_installer="dpkg"
+
+	if distro == 'fedora' or distro == 'rhel' or distro == 'suse':
+		package_installer="rpm"
+
 	# Make sure we can use dpkg on this system.
 	try:
-		subprocess.call( ['dpkg', '--version'], stdout=subprocess.PIPE )
+		subprocess.call( [package_installer, '--version'], stdout=subprocess.PIPE )
 	except:
-		sys.stderr.write( "Couldn't find dpkg, please update steamdeps for your distribution.\n" )
+		sys.stderr.write( "Couldn't find %s, please update steamdeps for your distribution.\n" % package_installer )
 		return False
 
 	return True
